0002-chunking.md

```diff
@@ -0,0 +1,271 @@
+
+Therefore, we will modify the permutation argument to use the constraints:
+```
+L_0(x) * (aggregation(x) - 1)
+L_{n-zk_rows}(x) * (aggregation(x) - 1)
+```
+and
+```
+(x - omega^(n-zk_rows+1)) *
+(x - omega^(n-zk_rows+2))
+(x - omega^(n-1))
+( aggregation(x)
+    * (w_0(x) + gamma + x * beta * shift_0)
+    * (w_1(x) + gamma + x * beta * shift_1)
+    * (w_2(x) + gamma + x * beta * shift_2)
+    * (w_3(x) + gamma + x * beta * shift_3)
+    * (w_4(x) + gamma + x * beta * shift_4)
+    * (w_5(x) + gamma + x * beta * shift_5)
+    * (w_6(x) + gamma + x * beta * shift_6)
+  -
+    * aggregation(x * omega)
+    * (w_0(x) + gamma + sigma_0 * beta)
+    * (w_1(x) + gamma + sigma_1 * beta)
+    * (w_2(x) + gamma + sigma_2 * beta)
+    * (w_3(x) + gamma + sigma_3 * beta)
+    * (w_4(x) + gamma + sigma_4 * beta)
+    * (w_5(x) + gamma + sigma_5 * beta)
+    * (w_6(x) + gamma + sigma_6 * beta) )
+```
+
+In addition, we should 'zero out' the `sigma_i` terms between `n-zk_rows+1` and `n-zk_rows+2`, to ensure that the randomness in the zk rows for `w_i` contributes to the aggregation.
+
+In total, this modified permutation argument achieves the same zero knowledge
+
+#### Number of zk rows
+
+The permutation argument interacts with the `c` chunks in parallel, so it is
+possible to cross-correlate between them to compromise zero knowledge. We know
+that there is some `c >= 1` such that `zk_rows = 2*c + k` from the above. Thus,
+attempting to find the evaluation at a new point, we find that:
+* the evaluation of every witness column in the permutation contains `k` unknowns;
+* the evaluations of the permutation argument aggregation has `k-1` unknowns;
+* the permutation argument applies on all but `zk_rows - 3` rows;
+* and thus we form the equation `zk_rows - 3 < 7 * k + (k - 1)` to ensure that we
+  can construct fewer equations than we have unknowns.
+
+This simplifies to `k > (2 * c - 2) / 7`, giving `zk_rows > (16 * c - 2) / 7`.
+We can derive `c` from the `max_poly_size` supported by the URS, and thus we find
+`zk_rows` and `domain_size` satisfying the fixpoint
+
+```text
+zk_rows = (16 * (domain_size / max_poly_size) + 5) / 7
+domain_size = circuit_size + zk_rows
+```
+
+### Technical approach in kimchi
+
+Support for chunked commitments already exists [in kimchi's supporting polynomial commitment library](https://github.com/o1-labs/proof-systems/blob/ca0b9e53d15e3ab5db6fb8c20617979cbf74215e/poly-commitment/src/srs.rs#L89). This RFC proposes to extend that support through to the kimchi prover, where an attempt to use the feature currently [results in an error](https://github.com/o1-labs/proof-systems/blob/ca0b9e53d15e3ab5db6fb8c20617979cbf74215e/kimchi/src/prover.rs#L170). This RFC also proposes to add support for chunked commitments and evaluations to pickles, where there is currently no support.
```

Comment:
I feel like we "buried the lede" here, this detail would be very helpful in RFC summary section